### PR TITLE
UIORGS-283: Create settings types

### PR DESCRIPTION
**Pull request by elsenhans**
https://issues.folio.org/browse/UIORGS-283

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/Settings/TypeSettings.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { Field } from 'redux-form';
+
+import { stripesShape } from '@folio/stripes/core';
+import { Select } from '@folio/stripes/components';
+import { ControlledVocab } from '@folio/stripes/smart-components';
+import TypeStatus from '../Utils/TypeStatus';
```
Comment (NikitaSedyx) [inline src/Settings/TypeSettings.js:8]: why is it called like a class? additionally we use named exports in our modules

File: src/Utils/TypeStatus.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default [
```
Comment (NikitaSedyx) [inline src/Utils/TypeStatus.js:1]: it's not an util, additionally it's not smth common as I understand

File: src/Settings/TypeSettings.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
```
Comment (NikitaSedyx) [inline src/Settings/TypeSettings.js:1]: please create a folder for this component

File: src/Utils/TypeStatus.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  { label: 'Active', value: 'Active' },
```
Comment (AndreiShumski2) [inline src/Utils/TypeStatus.js:2]: labels should be translated

File: src/Settings/TypeSettings.js (hunk 1 above)
Comment (elsenhans) [reply src/Settings/TypeSettings.js]: Thanks! Changed it to a named export.

File: src/Utils/TypeStatus.js (hunk 2 above)
Comment (elsenhans) [reply src/Utils/TypeStatus.js]: I moved typeStatus into the TypeSettings folder

File: src/Utils/TypeStatus.js (hunk 4 above)
Comment (elsenhans) [reply src/Utils/TypeStatus.js]: Thanks!The label are now translated.

File: src/Settings/SettingsPage.js [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { FormattedMessage } from 'react-intl';
 import { Settings } from '@folio/stripes/smart-components';
 
 import CategorySettings from './CategorySettings';
+import TypeSettings from './TypeSettings/TypeSettings';
```
Comment (AndreiShumski2) [inline src/Settings/SettingsPage.js:7]: please add `index.js` for exporting TypeSettings

**Review by AndreiShumski2 — APPROVED** [review]

File: src/Settings/SettingsPage.js (hunk 5 above)
Comment (elsenhans) [reply src/Settings/SettingsPage.js:7]: added index.js

File: src/Settings/TypeSettings/typeStatus.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+const typeStatus = [
```
Comment (NikitaSedyx) [inline src/Settings/TypeSettings/typeStatus.js:1]: export const

File: src/Settings/TypeSettings/index.js [hunk 7]
```diff
@@ -0,0 +1 @@
+export { default } from './TypeSettings';
```
Comment (NikitaSedyx) [inline src/Settings/TypeSettings/index.js:1]: please use named exports

**Comment by NikitaSedyx** [comment]
please add more coverage if possible, 80% is our gate

**Comment by elsenhans** [comment]
Sorry, but it is not possible to render the ControlledVocab component without mocking the ControlledVocab itself and just with mocking stripe-connect.

File: src/Settings/TypeSettings/index.js (hunk 7 above)
Comment (elsenhans) [reply src/Settings/TypeSettings/index.js]: its done.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,7 @@
 # Change history for ui-organizations
 
 ## (IN PROGRESS)
+* Create area Settings - Organizations - Types. Refs UIORGS-283.
```
Comment (NikitaSedyx) [inline CHANGELOG.md:5]: extra line

File: src/Settings/SettingsPage.js (hunk 5 above)
Comment (NikitaSedyx) [reply src/Settings/SettingsPage.js:7]: it's still not updated, please fix

Comment (NikitaSedyx) [reply src/Settings/SettingsPage.js:7]: it should be `import { TypeSettings } from './TypeSettings';` after your changes in export

File: CHANGELOG.md (hunk 8 above)
Comment (elsenhans) [reply CHANGELOG.md]: thanks, did remove the blank line

File: src/Settings/SettingsPage.js (hunk 5 above)
Comment (elsenhans) [reply src/Settings/SettingsPage.js]: sorry, my mistake :/
should be fixed now

**Review by NikitaSedyx — APPROVED** [review]
!important please use Squash instead of Merge to keep history clean

and wait for one more approve

**Comment by elsenhans** [comment]
PR is approved by @AndreiShumski2 and @NikitaSedyx 
Who else is goinig to review the PR?

**Review by usavkov-epam — APPROVED** [review]

**Comment by elsenhans** [comment]
Thank you @usavkov-epam !

**Pull request merged**